Blish HUD/GameServices/Contexts/FestivalContext.cs

```diff
@@ -0,0 +1,156 @@
+
+        /// <summary>
+        /// A Guild Wars 2 festival identified by the <see cref="FestivalContext"/>.
+        /// </summary>
+        public struct Festival {
+
+            private readonly string _name;
+            private readonly string _displayName;
+
+            /// <summary>
+            /// The unique identifying name of the festival.
+            /// </summary>
+            public string Name => _name;
+
+            /// <summary>
+            /// The localized display name of the festival.
+            /// </summary>
+            public string DisplayName => _displayName;
+
+            public Festival(string name, string displayName) {
+                _name        = name;
+                _displayName = displayName;
+            }
+
+            // Known festivals
+
+            /// <summary>
+            /// Guild Wars 2 Halloween Festival.
+            /// </summary>
+            public static readonly Festival Halloween = new Festival("halloween", Strings.Context_FestivalContext_Halloween);
+
+            /// <summary>
+            /// Guild Wars 2 Wintersday Festival.
+            /// </summary>
+            public static readonly Festival Wintersday = new Festival("wintersday", Strings.Context_FestivalContext_Wintersday);
+
+            /// <summary>
+            /// Guild Wars 2 Super Adventure Festival.
+            /// </summary>
+            public static readonly Festival SuperAdventureFestival = new Festival("superadventurefestival", Strings.Context_FestivalContext_SuperAdventureFestival);
+
+            /// <summary>
+            /// Guild Wars 2 Lunar New Year Festival.
+            /// </summary>
+            public static readonly Festival LunarNewYear = new Festival("lunarnewyear", Strings.Context_FestivalContext_LunarNewYear);
+
+            /// <summary>
+            /// Guild Wars 2 Festival of the Four Winds.
+            /// </summary>
+            public static readonly Festival FestivalOfTheFourWinds = new Festival("festivalofthefourwinds", Strings.Context_FestivalContext_FestivalOfTheFourWinds);
+
+            /// <summary>
+            /// Guild Wars 2 Dragon Bash Festival.
+            /// </summary>
+            public static readonly Festival DragonBash = new Festival("dragonbash", Strings.Context_FestivalContext_DragonBash);
+        }
+
+        private const string DAILY_GROUP_ID = "18DB115A-8637-4290-A636-821362A3C4A8";
+
+        private readonly Dictionary<int, Festival> _knownFestivalCategories = new Dictionary<int, Festival> {
+            { 79, Festival.Halloween },
+            { 98,  Festival.Wintersday},
+            { 162, Festival.SuperAdventureFestival },
+            { 201, Festival.LunarNewYear },
+            { 213, Festival.FestivalOfTheFourWinds },
+            { 233, Festival.DragonBash }
+        };
+
+        private CancellationTokenSource _contextLoadCancellationTokenSource;
+
+        private List<Festival> _activeFestivals = new List<Festival>();
```

Comment:
while it is technically possible that multiple festivals are active at once, theres no way anet will ever do this. for one, the decorations in cities would probably clip and they also stated that festivals are fillers between content releases. so i think they will never activate multiple festivals at the same time. 

what are your thoughts on providing just a single `_activeFestival`?

Reply:
I do agree that is is particularly unlikely, but with the increased number of "events" happening lately, I worry that "festivals" may begin to include some smaller events that potentially overlap.  I also get that it's less convenient not being able to check for just an individual festival.  Do the two new functions provided in my latest commit make you feel better about this?

Reply:
better :)